### ISSUE TITLE
RCS1228 should respect record type

### ISSUE THREAD

**Issue by iSeiryu** [JosefPihrt/Roslynator#750]
**Product and Version Used**: 
Roslynator.Analyzers 3.0.0

**Steps to Reproduce**:
https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-9#record-types
Create a record with param in xml doc and observe https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1228.md to be triggered:
```csharp
/// <summary>
/// Some user information
/// </summary>
/// <param name="Name">User's name</param>
public record User(string Name);
```
**Actual Behavior**:
param is flagged

**Expected Behavior**:
param should not be flagged for positional records

**Comment by Dryadxon** [comment]
This still happens with C# 10 `record struct` types using Roslynator.Analyzers 4.0.2

**Comment by JosefPihrt** [comment]
@Dryadxon fixed with a02cb1ca1483042e8b66c8867dac4639083d3fbe

**Comment by Dryadxon** [comment]
[Starting with C# 12](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12#primary-constructors), primary constructors will be allowed on normal structs and classes.
Since .NET 8 is not stable yet, it's not a major issue. I'm hoping there's a way to see if the type declaration has a primary constructor, but Roslyn doesn't seem to provide that kind of info.
